client/securedrop_client/sdk/__init__.py

```diff
@@ -147,12 +164,102 @@ def _send_json_request(
         if timeout:
             data["timeout"] = timeout
 
+        env = {}
+        if self.development_mode:
+            env["SD_PROXY_ORIGIN"] = self.server
+
+        # Streaming
+        if stream:
+            fobj = tempfile.TemporaryFile("w+b")
+
+            content_length = 0
+            filename = ""
+            sha256sum = ""
+
+            retry = 0
+            bytes_written = 0
+            download_finished = False
+
+            while not download_finished and retry < DOWNLOAD_RETRY_LIMIT:
+                try:
+                    # Open the process
+                    proc = subprocess.Popen(
+                        self._rpc_target(),
+                        stdin=subprocess.PIPE,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.PIPE,
+                        env=env,
+                    )
+
+                    # Send the data, included the range if we are retrying
+                    if retry > 0:
+                        data["headers"]["Range"] = f"bytes={bytes_written}-"
+
+                    data_str = json.dumps(data).encode()
+                    if proc.stdin is not None:
```

Comment:
I don't think `proc.stdin` should ever be None since we pass `stdin=subprocess.PIPE` into `subprocess.Popen`, so it should always be an object with `.write` and `.close`. But the linters complain without it, since it's possible for `Popen` objects to have `stdin` as None:

```
securedrop_client/sdk/__init__.py:202: error: Item "None" of "IO[bytes] | None" has no attribute "write"  [union-attr]
securedrop_client/sdk/__init__.py:203: error: Item "None" of "IO[bytes] | None" has no attribute "close"  [union-attr]
```

I guess the only case where it would be None is if there's some internal bug in the `subprocess` module. Still, it doesn't hurt to throw an exception. But which one? The only custom exceptions are RequestTimeoutError and ServerConnectionError and neither appropriate. I could make a new exception type, but that feels like more overhead than it's worth for a case that should never happen.